public/app/features/dashboard/panel_editor/QueriesTab.tsx

```diff
@@ -243,32 +254,44 @@ export class QueriesTab extends PureComponent<Props, State> {
         setScrollTop={this.setScrollTop}
         scrollTop={scrollTop}
       >
-        {isSharedDashboardQuery(currentDS.name) ? (
-          <DashboardQueryEditor panel={panel} panelData={data} onChange={query => this.onQueryChange(query, 0)} />
-        ) : (
-          <>
-            <div className="query-editor-rows">
-              {panel.targets.map((query, index) => (
-                <QueryEditorRow
-                  dataSourceValue={query.datasource || panel.datasource}
-                  key={query.refId}
-                  panel={panel}
-                  dashboard={dashboard}
-                  data={data}
-                  query={query}
-                  onChange={query => this.onQueryChange(query, index)}
-                  onRemoveQuery={this.onRemoveQuery}
-                  onAddQuery={this.onAddQuery}
-                  onMoveQuery={this.onMoveQuery}
-                  inMixedMode={currentDS.meta.mixed}
-                />
-              ))}
-            </div>
-            <PanelOptionsGroup>
-              <QueryOptions panel={panel} datasource={currentDS} />
+        <>
+          {isSharedDashboardQuery(currentDS.name) ? (
+            <DashboardQueryEditor panel={panel} panelData={data} onChange={query => this.onQueryChange(query, 0)} />
+          ) : (
+            <>
+              <div className="query-editor-rows">
+                {panel.targets.map((query, index) => (
+                  <QueryEditorRow
+                    dataSourceValue={query.datasource || panel.datasource}
+                    key={query.refId}
+                    panel={panel}
+                    dashboard={dashboard}
+                    data={data}
+                    query={query}
+                    onChange={query => this.onQueryChange(query, index)}
+                    onRemoveQuery={this.onRemoveQuery}
+                    onAddQuery={this.onAddQuery}
+                    onMoveQuery={this.onMoveQuery}
+                    inMixedMode={currentDS.meta.mixed}
+                  />
+                ))}
+              </div>
+              <PanelOptionsGroup>
+                <QueryOptions panel={panel} datasource={currentDS} />
+              </PanelOptionsGroup>
+            </>
+          )}
+
+          {enableTransformations && this.state.data.state === LoadingState.Done && (
```

Comment:
I think this should only hide if it is LoadingState.NotStarted -- the others should be OK no?

Reply:
or maybe put a spinner/ignore if NotStarted? but still show they exist?